### ISSUE TITLE
Please Create an Agent that scrapes scenes from Fagalicious.com/

### ISSUE THREAD

**Issue by CodyBerenson** [acvigue/pgma#13]
Fagalicious.com is similar to WayBig.com, in that it indexes scenes with coverart.  Many of the scenes that aren't on WayBig are on Fagalicious.  It would be terrific if there were a Fagalicious agent that functions similar to WayBig.  

Thanks in advance!

**Comment by JPH71** [comment]
Here is the scraper...

Fagalicious by the way does not give any director names... similar to waybig...

Thanks for helping me test..

[Fagalicious.bundle.zip](https://github.com/acvigue/pgma/files/4083963/Fagalicious.bundle.zip)

**Comment by JPH71** [comment]
I wish I knew where waybig and fagalicious source their coverart!

**Comment by CodyBerenson** [comment]
So, this site introduces some new complexities....quotation marks....longer than the normal dash....colons....

I'm testing different permutations....any thoughts:

https://fagalicious.com/nakedsword-originals-ricky-larkin-calvin-banks-bareback-the-gay-simple-life-day-1-big-rig-driving-challenge/

**Comment by JPH71** [comment]
i will most likely have to split the search query at the first problem
character
as i have designed it to look at subsequent result pages.. hopefully it
will find the title further down.
single apostrophes also cause issues...
let me know if you find any other problem characters or titles
i will make changes tomorrow when i get back from work...
off to bed now as its 01h33 here xxxx

cheers man

Jason

**Comment by JPH71** [comment]
just tested at least one can replace the apostrophes and colons with null,
and the search works
for the long dashes i will have to split

**Comment by JPH71** [comment]
Does not work for long dashes... So will split at first occurrence of theses

**Comment by j-ktz** [comment]
Awesome! I would use this scraper too. Let me know if I can help test? Thanks!!

**Comment by JPH71** [comment]
Yes please help with testing...
All the help is much appreciated 👍

**Comment by j-ktz** [comment]
Sure, new to the Github world, so might need some instruction. But I just tested one file and it didn't match. Tried some naming variations (curly quotes vs straight quotes), removing hyphen after studio name/keeping hyphen after studio name, etc. Here's the log if it helps. I have some other files so if you want me to test anything specific, let me know. Happy to help!

[com.plexapp.agents.fagalicious.log](https://github.com/acvigue/pgma/files/4084038/com.plexapp.agents.fagalicious.log)

**Comment by CodyBerenson** [comment]
thanks my friend

**Comment by JPH71** [comment]
Hi Cody
Here is the second version of fagalicious which hopefully sorts out the colons, curly apostrophes and quotation marks..

Hey J-KTZ
With all the scrappers I have written avoid using curly apostrophes, em-dashes, and replace colons with " - " etc.
though it doesnt matter much in the fagalicious bundle as the bundle defines commas, all apostrophes and all quotation marks as invalid and uses only the site entry upto the first invalid character

[Fagalicious.bundle.zip](https://github.com/acvigue/pgma/files/4088247/Fagalicious.bundle.zip)

**Comment by CodyBerenson** [comment]
Ok, i'll do some testing and let you know.  

xoxoxo

**Comment by JPH71** [comment]
Thanks man
Going to start on makeflixxtv next then hopefully gayVodClub after that..

**Comment by CodyBerenson** [comment]
I'm having mixed results.  Using the rule above (one can replace the apostrophes and colons with null) and assuming we leave the long dashes, i had one file match and two do not.  Stupid question:  what should the title for this be:

https://fagalicious.com/nakedsword-originals-ricky-larkin-calvin-banks-bareback-the-gay-simple-life-day-1-big-rig-driving-challenge/

**Comment by JPH71** [comment]
Rename it as such
(NakedSword Originals) - Ricky Larkin fucks Calvin Banks bareback in "The
Gay Simple Life - Day 1 - Big Rig Driving Challenge" (2020)
With the fagalicious scrapper the resultant search query will be
NakedSword Originals Ricky Larkin fucks Calvin Banks bareback in It will
return two results and match against one of them.
This was the movie title that I used to test the scrapper..
Removal of curved apostrophes, em dashes etc gives the waybig scrapper a
chance to match it too.

Hope this helps
Jason

**Comment by acvigue** [comment]
remember when you are submitting the finalized agent to send a PR. check the CONTRIBUTION.md file for instructions

**Comment by j-ktz** [comment]
Tried testing 3 files, 2/3 matched/ Here is the filename vs. the names on Fagalicious of the one that didn't match with the log. Hopefully this helps: 
<img width="813" alt="screenshot 2020-01-20 at 8 26 55 PM" src="https://user-images.githubusercontent.com/59346628/72768181-82a98d00-3bc4-11ea-939a-aef70c77f227.png">
<img width="600" alt="screenshot 2020-01-20 at 8 25 46 PM" src="https://user-images.githubusercontent.com/59346628/72768183-86d5aa80-3bc4-11ea-96e9-f35f17a5607c.png">
[com.plexapp.agents.fagalicious.log](https://github.com/acvigue/pgma/files/4088782/com.plexapp.agents.fagalicious.log)

**Comment by CodyBerenson** [comment]
Jason, windows doesn't allow " in a file name.  I'm so confused.  

On a better note, I am pretty sure I submitted a valid Pull Request for the GEVI scraper.  If that worked, we'll get the rest of the scrapers updated.

**Comment by CodyBerenson** [comment]
Jason, my pull request worked and Aiden added GEVI to the repository!    Fagalicious isn't ready, but when you get a moment please upload the rest here and I'll create pull requests.  :)

**Comment by JPH71** [comment]
As I keep repeatin, curved apostrophes, curved quotation marks, em-dashes
are quite problematic ... this is how the scrapper rendered your video
(Falcon Studios) - Skyy Knox and Ethan Chase flip fuck in Love and Lust in
Montreal (2019)..... in the log file the  with a ? inside it... as you
can see even notepad and gmail..hyml are showing it differently.... stick
to straight apostrophes and quotes from now on...

**Comment by j-ktz** [comment]
Strange, I used straight quotes in the filename, not sure why they would come out as curved...but thank you. Sorry.

**Comment by JPH71** [comment]
Cody it does allow " them in file names... possibly your windows is not
allowing it as its a unicode character it does not like... try typing it
straightoff the keyboard or in notepad first making sure notepad is in
ASCII mode...

**Comment by CodyBerenson** [comment]
Jason, Windows will not allow quotes in filenames.  When created in notepad and copied into windows explorer, windows strips the quote from the filename.  won't allow it.  period.  If the site requires the scraper to use a quote, anyone with a windows plex media server won't be able to use the scraper.

**Comment by JPH71** [comment]
type the name first in notepad... make sure notepad is in ASCII or better
still use notepad++

if you remember, the other day when i made a correction for one of your
titles, a normal dash got converted to a long dash and thus woldnt work...
Cody ended pointing it out to us...

I dont know why windows does this....

No worries..

you can see this in the log here...

2020-01-20 20:32:46,209 (2cfd8fd4f0) :  INFO (sandbox:19) - Fagalicious -
SEARCH::      ->regex - ^\((?P<studio>.+)\) - (?P<title>.+)
\((?P<year>\d{4})\)
2020-01-20 20:32:46,210 (2cfd32d4f0) :  INFO (sandbox:19) - Fagalicious -
SEARCH:: media.title - Falcon Studios Skyy Knox and Ethan Chase Flip Fuck
in love and Lust in Montreal
.

**Comment by acvigue** [comment]
lordy jesus this is why i use linux

**Comment by JPH71** [comment]
From the olden days since DOS and through to windows quotes are valid
characters to use in filenames... It works perfectly on my Windows 10 system
here is a snip I just took
Something is not right on your system.

**Comment by acvigue** [comment]
@JPH71, please commit the code you have and send a pull request and i will take a look. follow CONTRIBUTING.md

**Comment by JPH71** [comment]
lol... Aiden you are hardcore!

**Comment by JPH71** [comment]
I will do it with the next scrapper - Cody stepped in and helped with the
fagalicious bundle.
I am off to bed now as its 02h28 here....
Thanks for all the help.

**Comment by CodyBerenson** [comment]
Jason, there's no snippit in the above message.  Regardless, I don't want to argue with you.  

https://en.wikipedia.org/wiki/Filename

I appreciate you creating the bundle.  Looks like in its current state it just won't be one that I can use.

**Comment by acvigue** [comment]
problems like this are why i just suck it up and make an agent for each site (falcon studios in tyis case)

**Comment by JPH71** [comment]
Hey Cody

I sent the last message by mistake with Gmail and it seems that the system strips all attachments.here is the snippet
![Capture](https://user-images.githubusercontent.com/43102249/72771012-78ed4d00-3bf7-11ea-8510-9aa5b0b629f8.PNG)
hopefully you can see it now

**Comment by JPH71** [comment]
As I said you are Hardcore!
I am also going to try with the AEBN website as I have a feeling they may
discontinue the old format one... Especially now I know how to use the
developer console to test my xpath ....

**Comment by CodyBerenson** [comment]
Jason, I thought maybe I was crazy.  I googled Windows Quote Marks in Filenames.  Forbidden.  Straight out of Microsoft's Windows Documentation:

Use any character in the current code page for a name, including Unicode characters and characters in the extended character set (128–255), except for the following:

The following reserved characters:

< (less than)
> (greater than)
: (colon)
" (double quote)
/ (forward slash)
\ (backslash)
| (vertical bar or pipe)
? (question mark)
* (asterisk)

Perhaps not in your particular install somehow...or whatever VM, emulator, etc.  Doesn't matter.  If the site requires quote marks, then I can't use the Agent.

**Comment by JPH71** [comment]
Do single quote marks work on your system?

Maybe replace the double quotes with single quotes... They will serve the
same purpose as double quotes in the scrapper...
Also try using notepad++ to write out your filenames first and see if
windows strips the quotes of them...
I have never being on a system where I have seen the issue you are
experiencing especially since windows brought out long filenames...

Jason

**Comment by JPH71** [comment]
I am using plain old windows 10... Before that I was on windows 7... Hence
why I am so surprised about this issue...

At least the scrapper will work for files with no quotes...

And j-ktz also uses quotes in his filenames... I wonder what version of
Windows he's using...

**Comment by CodyBerenson** [comment]
The single quote is the apostrophe.  I tried apostrophes which didn't work, but I can certainly try again. 
  
LOL.  You are hilarious.  The issue that I am experiencing is by design, per windows documentation worldwide.  LOL.

**Comment by JPH71** [comment]
So both speech marks " and apostrophes ' don't work...

I will change the scrapper so that you can use back ticks instead of single
or double quotes...

I have worked on computers in 5 countries and have never come across this
issue hence why I confidently said what I said...

**Comment by CodyBerenson** [comment]
LOL.  I'm completely just teasing with you.  I am not about to compare our IT skills.  LOL.  there's a reason why one of us is the developer, and one is the tester.  ;)  sleep well, prince.

**Comment by CodyBerenson** [comment]
Here's what I don't understand though.  I've only gotten one match so far, and it had quotes on the site and i used no quotes in the filename which worked just fine:

https://fagalicious.com/familydick-daddy-lessons-chapter-4-sit-back-relax-taylor-reign-donnie-argento/

(FamilyDick) - Daddy Lessons – Chapter 4 Sit Back & Relax (2020).mp4

**Comment by JPH71** [comment]
That is bloody strange as everytime when I took out the quotes it wouldn't
work... Hence why I split the search at the very first quote...

Could you establish a matrix of search rules that I can then implement...

**Comment by acvigue** [comment]
Can't work on it now, but in one of the agents, I have google searching all links of a site that match 95% or more of the search string. You could try to adapt that code. If I could get this agent PRed as-is, I can edit it.

**Comment by CodyBerenson** [comment]
Because that one file worked and none of the others did, I wasn't sure that I was using the correct studio name, hence me asking what I should name the file.  I'm going to try with another file, that isn't familydick nor nakedsword.

**Comment by CodyBerenson** [comment]
Also Jason, when you get a chance, can you upload your final versions of AEBNii, HomoActive, WayBig, GayDVD, GayHot.  I will upload, and if there are differences between the Master, I'll open a pull request with Aiden to merge into the master....so that anyone venturing to the site has amazing agents.

Aiden, I'm going to test out the studio-specific sites, but am willing to bet that most don't work.  I'll let you know which ones don't and in my opinion removed from the master.....and then you or Jason or whoever can (if so inclined) fix them, and then we'll get them back there.  Not sure its worth the effort though, since Studios change their site and break the index.

**Comment by CodyBerenson** [comment]
lol.  So, somehow this worked.  I replaced the curly single and double quotes with standard apostrophe:
Filename:  (Men.com) - Theo Brady rides Ryan Bones' thick, raw cock in 'Caught in Mom's Panties' (2020)

Link:  https://fagalicious.com/men-com-theo-brady-ryan-bones-bareback-caught-in-moms-panties/

As Presented on Site:  Men.com: Theo Brady rides Ryan Bones’ thick, raw cock in “Caught in Mom’s Panties”

I'll upload my suggested rules matrix.

**Comment by CodyBerenson** [comment]
Suggested rules:
(Studio) - Filename (YYYY).ext

Replace colon, question-mark, asterix with null
Replace any quote (double, curly) with standard apostrophe 
Replace any dash (en-dash, em-dash) with standard hyphen
Leave parentheses as is

**Comment by CodyBerenson** [comment]
Jason, stupid question:  since I haven't gotten a single NakedSword file to work, can you verify that the correct studio is (NakedSword Originals)

https://fagalicious.com/nakedsword-originals-ricky-larkin-dakota-payne-bare-hookups-and-fuck-downs/

**Comment by JPH71** [comment]
Hi Cody
Sorry I was offline yesterday evening...
We had no bloody internet..
Hopefully it will be up today and running x

**Comment by JPH71** [comment]
> Jason, stupid question: since I haven't gotten a single NakedSword file to work, can you verify that the correct studio is (NakedSword Originals)
> 
> https://fagalicious.com/nakedsword-originals-ricky-larkin-dakota-payne-bare-hookups-and-fuck-downs/

Hi Cody 

This one worked - I used NakedSword Originals - here is the scrapped image

![image](https://user-images.githubusercontent.com/43102249/72939735-5d07b980-3d65-11ea-93df-ad79d1e007a7.png)

And the movie properties
![image](https://user-images.githubusercontent.com/43102249/72940003-eb7c3b00-3d65-11ea-8422-d8c7aa521e5b.png)


Filename: (NakedSword Originals) - Ricky Larkin bangs Dakota Payne in “Bare – Hookups And Fuck Downs” (2019)

I let in the emdash and the quotes as the scrapper would just ignore them.

I know you can't enter quotes on your system, so I am going to add back ticks ` to the list of invalid characters and every time you have a quote in your filename you can replace it with a backtick or two back ticks so that it looks like a quote...

**Comment by JPH71** [comment]
> Suggested rules:
> (Studio) - Filename (YYYY).ext
> 
> **Replace colon, question-mark, asterix with null**
> Replace any quote (double, curly) with standard apostrophe
> Replace any dash (en-dash, em-dash) with standard hyphen
> **Leave parentheses () brackets [] {} as is**
> 
> Clearly, any particular site may make some of the suggested rules moot.

The changes in bold are already implemented..
I tried replacing the quotes with apostrophes but the search objects to the use of both and woldn't return owt when these were used, so for the movie
_NakedSword Originals: Ricky Larkin bangs Dakota Payne in “Bare – Hookups And Fuck Downs”_ 
the scrapper searches on 
_NakedSword Originals: Ricky Larkin bangs Dakota Payne in_ 

on the hard disk the filename is stored as
_(NakedSword Originals) - Ricky Larkin bangs Dakota Payne in “Bare – Hookups And Fuck Downs” (2019)_

In your case you can put 2 backticks to look like quotes or 2 apostrophes if your system allows... as the scrapper considers both of these as invalid characters in the search string you will end up with the same search string

**Comment by JPH71** [comment]
in the last email - i made a slight mistake, the scrapper no longer uses the studio name in the search. The rest remains the same...

here is the updated scrapper
[Fagalicious.bundle.zip](https://github.com/acvigue/pgma/files/4100523/Fagalicious.bundle.zip)

Could you create issues for MaleFliXXXTV and GayVODClub as they are the next scrappers I want to write.

Thanks for all your help

**Comment by CodyBerenson** [comment]
I'm a little confused by your last few posts, but I'll give it a try and ask for clarification if necessary.

New Enhancement Requests have been opened for MaleFliXXXTV and GayVODClub.  You should see them as the two most recent issues.

**Comment by CodyBerenson** [comment]
It did work for that one scene, but not 4 of the others.  Should we just consider Fagalicious a wash and move on, or would you like the 4 scenes that don't work for troubleshooting?

**Comment by JPH71** [comment]
Yes please send the four scenes.
It has worked for me 8 out of 8 times ... However such a disparity in
results is absurd..
It would be great to get it working as it has loads more than waybig as it
has being running longer...

**Comment by CodyBerenson** [comment]
ok.  :). im Willing to work at it as much as you’d like, I was just worried about your time.

https://fagalicious.com/nakedsword-originals-parker-payne-alam-wernik-rags-to-riches-episode-2-the-upper-west-side/

https://fagalicious.com/nakedsword-originals-dante-colle-josh-moore-calvin-banks-bareback-scared-stiff-2-the-amityville-whore-scene-1-9-5-inches-well-take-it/

https://fagalicious.com/nakedsword-originals-jason-vario-pierce-paris-rikk-yorks-threeway-making-rent/

https://fagalicious.com/nakedsword-originals-ricky-larkin-calvin-banks-bareback-the-gay-simple-life-day-1-big-rig-driving-challenge/

ps wanted to reinforce how good your WayBig Index agent is....of over 1150 men.com scenes i have indexed so far only about 3 didnt match and i had to manually create metadata    KUDOS!

**Comment by CodyBerenson** [comment]
gentle reminder, when you have a moment, upload all of your final agents here and ill create pull requests for aiden to merge into the master. :)

**Comment by acvigue** [comment]
send me the fagalicious code in zip i don’t care but i would like to try something.

**Comment by JPH71** [comment]
Oooops I will do that...

**Comment by JPH71** [comment]
I will upload all the code in a moment...
Fagalicious is a bit weird... Let me explain it as such...
I search for Studio + title
The scrapper finds the usual quote so trims the title
The result is that it finds 2 results one of which is the correct...
First result it rejects as it's not a match... It goes to the second one
which is a match but the scrapper till matches it against the first
result... It's as if the scrapper did not iterate...

In the second incident I took the studio name out of he search string ...
Got the same two results but with the matching result as first in iteration
so got a match... Could you look into this too...
The title was the one Cody mentions above with Larkin and Dakota...

**Comment by JPH71** [comment]
here are all the bundles
[HomoActive.bundle.zip](https://github.com/acvigue/pgma/files/4105974/HomoActive.bundle.zip)
[WayBig.bundle.zip](https://github.com/acvigue/pgma/files/4105975/WayBig.bundle.zip)
[AEBNii.bundle.zip](https://github.com/acvigue/pgma/files/4105976/AEBNii.bundle.zip)
[Fagalicious.bundle.zip](https://github.com/acvigue/pgma/files/4105977/Fagalicious.bundle.zip)
[GayDVDEmpire.bundle.zip](https://github.com/acvigue/pgma/files/4105978/GayDVDEmpire.bundle.zip)
[GayHotMovies.bundle.zip](https://github.com/acvigue/pgma/files/4105979/GayHotMovies.bundle.zip)
[GEVI.bundle.zip](https://github.com/acvigue/pgma/files/4105980/GEVI.bundle.zip)

**Comment by CodyBerenson** [comment]
Jason, many thanks. I’ll initiate a pull request. :)

**Comment by JPH71** [comment]
Cheers

**Comment by CodyBerenson** [comment]
Jason, pull request successful.  All of the updated/new agents (sans Fagalicious) are now merged into the Master code.  :)  Well done, my friend.

**Comment by CodyBerenson** [comment]
Jason, I've run across a bug for the Fagalicious files that do match.  The site lists the date, for example as 1/17/20, yet the metadata populated into plex for all of the matches is 12/31/20.  here's an example:

https://fagalicious.com/men-com-kaleb-stryker-theo-brady-bareback-jizzed/

![image](https://user-images.githubusercontent.com/40373801/73092092-e32d1880-3ea9-11ea-8261-e4e00f118e08.png)

**Comment by JPH71** [comment]
I will look into that...
It's supposed to default to the last day of the year if it fails to find
the date ..
Obviously there is something wrong in the code..
Thanks for finding that

**Comment by JPH71** [comment]
Hi Cody

I have made the changes as requested and sorted out the date issue.

On another note Windows 10 today decided not to allow me to type in a double-quote, saying it was an illegal filename character. On closely looking at my filenames which had quotes I realised that they were not straight up and down quotes **"** but slanted to around 1 O'clock **“** - typical word smart opening quotes

my apologies to you and J-KTZ - though i still dont know how i have gotten away with this for years.... and why my system has now started whingeing about it....

For this file:
https://fagalicious.com/familydick-daddy-lessons-chapter-4-sit-back-relax-taylor-reign-donnie-argento/

As it started with an illegal character, the search system seemed fine and found it without the quotes, however if one put the quotes in whilst manually searching it failed as per expectations. 

In the scrapper the file with quotes in would have failed to search as it would not have a search title as it started with an illegal character, now it searches for on the title upto the next illegal character.

I also made changes on retrieving background art for this video as it failed, this video stored its other images in a different format- so i catered for this.

In relation to the files you could not scrape. I have made some more changes to get them to work, this had to deal with iterating trough the returned lists of titles after a search had being executed. good lessons learnt from this

This was how I named your files:
_(NakedSword Originals) - Jason Vario, Pierce Paris and Rikk York’s hot threeway in “Making Rent” (2018)
(NakedSword Originals) - Dante Colle and Josh Moore tag team Calvin Banks bareback in “Scared Stiff 2 - The Amityville Whore, Scene 1 - 9.5 Inches We’ll Take It!” (2019)
(NakedSword Originals) - Parker Payne fucks Alam Wernik in “Rags to Riches, Episode 2 The Upper West Side” (2019)
(NakedSword Originals) - Ricky Larkin fucks Calvin Banks bareback in “The Gay Simple Life – Day 1 - Big Rig Driving Challenge” (2020)
(FamilyDick) - “Daddy Lessons – Chapter 4 Sit Back & Relax“ (2020).mp4_

And here is how they look on Plex!
![image](https://user-images.githubusercontent.com/43102249/73128244-31ced580-3fc4-11ea-8429-293f23cec2ee.png)

Cody: Here is the zip, could you please do your magic and create the pull request stuff on it?
[Fagalicious.bundle.zip](https://github.com/acvigue/pgma/files/4112991/Fagalicious.bundle.zip)

**Comment by j-ktz** [comment]
Thanks @JPH71. Does that mean files in Plex that I've had with quotes will work if I download the new bundle? Or do I need to do something on my end? Who knew quotes would be my downfall lol.

**Comment by CodyBerenson** [comment]
J-ktz:  Happy weekend!

Jason:  Happy weekend to you as well!  I'll give the new bundle a thorough test a little later and report back.  Thanks for all of your hard work!  

Cheers my friends!

**Comment by JPH71** [comment]
try the the new bundle and look through the relevant logs

**Comment by CodyBerenson** [comment]
Jason:  Well done!!!  Nearly my entire NakedSword library matched with your update!  The few files that didn't all follow similar naming conventions, probably not planned for in your codeing of the scraper.  For me, 90% is good enough, but I know you're a bit more of a perfectionist, so here's the representative sample of what wouldn't match.  Definitely different naming than most of the files on here:

https://fagalicious.com/nakedsword-originals-bare-devin-franco-jason-vario-bareback/

Really appreciate your diligence and your patience!

**Comment by JPH71** [comment]
I like it that you are able to find the ones that don't work...
I got nowt to do today... And walking outside in the cold is not really
floating my boat...
So a-coding I will do...

**Comment by JPH71** [comment]
It worked for me... took a bit longer as it had to go through 526 search result pages....

![image](https://user-images.githubusercontent.com/43102249/73134510-1ea51f80-402f-11ea-9669-75a5658ac1d7.png)

i think i am going to make a change to the code - so that it does a search against whole alphabetic words of the title - or if the search string ends up being one word... search against the second split....

let me think about it....

**Comment by JPH71** [comment]
Hi Cody

I have made some changes to the scrapper... found out that search query can be no longer than 50 characters...

I tested it with the previous problem files and the 2 new ones:
This was how I named your files:
**_(NakedSword Originals) - Jason Vario, Pierce Paris and Rikk York’s hot threeway in “Making Rent” (2018)
(NakedSword Originals) - Dante Colle and Josh Moore tag team Calvin Banks bareback in “Scared Stiff 2 - The Amityville Whore, Scene 1 - 9.5 Inches We’ll Take It!” (2019)
(NakedSword Originals) - Parker Payne fucks Alam Wernik in “Rags to Riches, Episode 2 The Upper West Side” (2019)
(NakedSword Originals) - Ricky Larkin fucks Calvin Banks bareback in “The Gay Simple Life – Day 1 - Big Rig Driving Challenge” (2020)
(FamilyDick) - “Daddy Lessons – Chapter 4 Sit Back & Relax“ (2020)_
(NakedSword Originals) - “Bare – Devin Franco and Jason Vario” (2019)
(NakedSword Originals) - Alam Wernik, Nic Sahara, Jay Dymel, Blake Ryder and Sean Duran in “Five Brothers - The Takedown – Episode 4, Brother’s Gang Bang” (2019)**

her is the screenshot:
![image](https://user-images.githubusercontent.com/43102249/73135840-e7d70580-403e-11ea-8d22-0dbbfd74afb5.png)

here is the bundle
[Fagalicious.bundle.zip](https://github.com/acvigue/pgma/files/4113491/Fagalicious.bundle.zip)

**Comment by CodyBerenson** [comment]
Jason:  Perfect!  Resolved all 5 of the unmatched files!  You're genius!

**Comment by CodyBerenson** [comment]
p.s. This is the easiest of the scrapers to use.  Its a straight copy and paste from the site, add the (year) and parens/hyphen around the studio name.  no manipulating apostrophes, etc.  

Gold Medal to Jason!

**Comment by JPH71** [comment]
Perfect...
I now need to sort out the rest of my Tim tales videos that waybig couldn't
scrape and try them on here...
Working on gayVodclub now
Xxx

**Comment by CodyBerenson** [comment]
LOL.  you mean all the Tim Tales that i was too impatient to wait for a scraper and manually meta-data'd them all!  LOL

Working on my Kristen Bjorn collection now.  40 out of 40 matches so far.  perfect 100%

**Comment by JPH71** [comment]
Waybig works well with timetales
However not all timtales are on it...
Hoping they will be on fagalicious as it's being running longer

**Comment by CodyBerenson** [comment]
Have you noticed any issue with Waybig not pulling in covers?  it may be a particular period of scenes (i seem to be in a 2012 pocket right now)....but, it pulls in metadata perfectly, just not the cover.  

Can you give it a try?

https://www.waybig.com/blog/2012/03/07/big-dicks-at-school-paying-my-way-through-school-ridge-michaels-marc-dylan/

**Comment by j-ktz** [comment]
Yeah, I've noticed that Cody. It seems to be older titles that have a poster in a different aspect ratio. I usually just go and copy the URL and paste directly into Plex since everything else comes in fine.

**Comment by JPH71** [comment]
i will look into that, as i need to put some more work into waybig.
give me a couple of days x

**Comment by JPH71** [comment]
and send me a few more offending titles, so that i can look at their html..

**Comment by CodyBerenson** [comment]
I'm nearing the end of my Men.com collection.  between WayBig and Fagalicious, fantastic matching.  :)

Here's another that pulls the metadata but not the coverart:

https://www.waybig.com/blog/2011/11/14/drill-my-hole-disturbing-the-peace-rafael-alencar-devin-moss/

**Comment by JPH71** [comment]
Cheers for that x

**Comment by JPH71** [comment]
Here you go, tested it against the 3 drill my hole files...

![image](https://user-images.githubusercontent.com/43102249/73222432-1f3ad480-415b-11ea-9819-edc2c5328090.png)


[WayBig.bundle.zip](https://github.com/acvigue/pgma/files/4119820/WayBig.bundle.zip)

**Comment by CodyBerenson** [comment]
cheers!

I'll open a pull request for the updated WayBig and the brand new Fagalicious.  Have you made any changes to it or is the latest code here?

**Comment by JPH71** [comment]
i was looking through the plex developers's manual yesterday and noticed
that one can force the agent to redownload all its metadata so i was
thinking of introducing this preferences into the scrapers... so that
previously matched films can be be forced matched when a change occurs...

What do you think?

**Comment by CodyBerenson** [comment]
when you say "a change occurs" do you mean on the scraper-site side?  (i.e. sometimes i find coverart on fagalicious that i like better than what was already matched on waybig (or vice versa), and i'll change the filename to match the specific site/agent requirements, and the change is picked up/new match occurs the next time the library scans...

So, without full context...this functionality sounds dangerous.  LOL!  

Here's why:  This may be a limitation with Fagalicious.com itself.  If you throw too many files at it simultaneously, the site locks you out momentarily.  I tend to drop and match in groups of 20ish, and for the last 8 or so i have to manually rematch after a minute or two passes.  

However, the temporary freeze-out i can live with.  I am now working on my NextDoor collection, so I have WayBig and Fagalicious side-by-side.  At least for NextDoor, Fagalicious has many, many more files indexed....and the coverart on Fagalicious mostly (not always ) fits into Plex better than the same file's coverart on WayBig.  And, the Fagalicious has one less step when renaming a file (changing curly single quote to a standard apostrophe).  

So, re: redownloading all metadata, i'll leave that thought and decision up to you.  Re: perservering with Fagalicious when I wanted to throw in the towel, I am HUGELY grateful to you.  The combination of the two makes scene scraping ideal.  

Finally, if there's anything you can do to get these porn sites to index/coverart ALL scenes from every studio, that would be optimal.  LOL!  LOL! For favorite studios such as TimTales, and Pride Studios, because I want every scene indexed, I've ended up having to be creative with coverart when scenes aren't indexed....(getting the metadata isn't an issue because of IAFD)

I was halfway going to ask you to think about creating a scene scraper for GEVI that just doesn't return the coverart and use what Plex will come up with during a scan (which would be the equivalent of all of the site specific scrapers that have no DVD-looking coverart)...but, let's leave that on the low list of priorities for now.   

ok, i'm initiating the pull request for the WayBig fix and the new Fagalicious.  

Questions:  Are you going to add "Developed Phenomenal Porn Scrapers" to your CV?

**Comment by CodyBerenson** [comment]
ok my friend, i created a pull request to merge Fagalicious and the update to WayBig into the master code.  Once Aiden does his part, we'll have to move on to a different issue to chat against.  ;)

**Comment by JPH71** [comment]
When you scrape for a movie it brings down the metadata
With waybig it only brought down the poster and not the background.
While I was reading the Plex developer pdf there was a section that said
you could have a setting so that when you refreshed an item it would act
like it was a new item and reset everything.
I was thinking of putting that in the preferences of the scrappers..
Now you have mentioned the freezing if the sites when you throw to much, I
will put in a delay request too in the preferences to try and counter that.

My CV would be unique in the least 😆

Cheers x

**Comment by CodyBerenson** [comment]
A delay request.  i like that.  i'll hold on on indexing my falcon and raging stallions scenes...they're ready to go but with the volume of them, it will be a good test for the delay.

**Comment by JPH71** [comment]
Perfect...
I will start on that now..
You mentioned something about GEVI scenes...
Let's put that in another thread and I will take a look at that
Xx

**Comment by JPH71** [comment]
I will have to work on it tomorrow... No bloody twatty internet again...

**Comment by JPH71** [comment]
Here's fagalicious with delay implemented.
it is a choice of 0, 20, 40 or 60 seconds
[Fagalicious.bundle.zip](https://github.com/acvigue/pgma/files/4137802/Fagalicious.bundle.zip)

**Comment by JPH71** [comment]
Here's waybig with delay implemented.
it is a choice of 0, 20, 40 or 60 seconds

[WayBig.bundle.zip](https://github.com/acvigue/pgma/files/4137819/WayBig.bundle.zip)

**Comment by CodyBerenson** [comment]
hey my friend.  for me, the results were intermittent.  I had a new library of 32 files go through beautifully.  then i had to fight for quite a while with a new library of 127 files.  some matched.  others i had to unmatch, match, fix match, unmatch, match.  at one point i reverted back to the old code which worked fine (as long as i didn't go to fast and time out the site).  Other than the timer, did you make any other changes to the code?  

For all i know, it could just be my system.  i'll keep trying.

**Comment by CodyBerenson** [comment]
Confirmed.  I'm reverting back to the previous version.

**Comment by CodyBerenson** [comment]
waybig is down for maintenance.....i wonder if they're going to break the scraper!  LOL

**Comment by JPH71** [comment]
Hi Cody

Here is a link detailing the differences in the code. As yu can see there
is nowt in there that adversely changed the way Waybig operates.
The default delay is still set to zero - unless you change the parameters.

LINK: https://www.diffchecker.com/8js0EtwQ

primary provider was set to true as i realised when trying to
change parameters that waybig did not show up on the main list of agents -
one had to drill through the Gay Adult agent to get to it..

And of course there is the added background art...

This difference is between now and the 23rd of January

**Comment by JPH71** [comment]
Hi Cody

I found the error. It was in the setting up of the preferences.
I had left the seconds in string format rather than numerical.
here are the latests zips.
[Fagalicious.bundle.zip](https://github.com/acvigue/pgma/files/4142161/Fagalicious.bundle.zip)
[WayBig.bundle.zip](https://github.com/acvigue/pgma/files/4142162/WayBig.bundle.zip)

**Comment by CodyBerenson** [comment]
Hmmmmm...unable to open the settings for fagalicious or WayBig, in either Chrome or firefox....

Any thoughts?

**Comment by CodyBerenson** [comment]
So, without being able to change the settings I tested it against a library of 83 files (77 from Fagalicious, 6 from Waybig).  ALL 83 MATCHED.  I've never seen performance from the Agents like it is now.  there was no lockup on Fagalicious.  So, i'm not even sure it needs the timer.  lol.  

I'm going to test now against a new library with 134 files in it.  You ROCK!!!!!!

**Comment by JPH71** [comment]
I must have screwed something else up...
I will compare their settings against homoactive and the rest..
Then I will send them back to you.

**Comment by JPH71** [comment]
In the meantime if you need to change the default time... Open the JSON
file and change it there.

**Comment by JPH71** [comment]
Dear Cody

I have finally found the error that prevented one from editing the default preferences for waybig and fagalicious.

Whilst correcting this error I realised that the all the agents I created showed up on the main agents screen alongside the main Gay Adult agent as if they were primary providers as well us being below it. I have corrected this and no the agent screen looks much neater.
![image](https://user-images.githubusercontent.com/43102249/73986329-3df85280-4935-11ea-8529-31f9ac1cb7c1.png)

I am including all the zips for all the agents in this email. Could you do your magic in relation to the pull requesty thingy...

I am still waiting on Aiden in relation to setting cookies, but I guess he must be busy.

Hope you are fine and dandy... hope to hear your thoughts about this soon.

Kind Regards

Jason
[GEVI.bundle.zip](https://github.com/acvigue/pgma/files/4168177/GEVI.bundle.zip)
[HomoActive.bundle.zip](https://github.com/acvigue/pgma/files/4168178/HomoActive.bundle.zip)
[WayBig.bundle.zip](https://github.com/acvigue/pgma/files/4168179/WayBig.bundle.zip)
[AEBNii.bundle.zip](https://github.com/acvigue/pgma/files/4168181/AEBNii.bundle.zip)
[Fagalicious.bundle.zip](https://github.com/acvigue/pgma/files/4168183/Fagalicious.bundle.zip)
[GayDVDEmpire.bundle.zip](https://github.com/acvigue/pgma/files/4168184/GayDVDEmpire.bundle.zip)
[GayHotMovies.bundle.zip](https://github.com/acvigue/pgma/files/4168185/GayHotMovies.bundle.zip)

**Comment by JPH71** [comment]
Just another thought in relation to the agents: What is everyone's thoughts about splitting the Gay Adult Agent into three?
One for searching through gay blogs, one for searching through Gay VOD sites like AEBN and one for Studios..
As the number of agents increases so does the time and the number of http requests sent out by our plex servers... One could then just have a libraries agent set to either blog, vod or studio depending on the contents of one's library.....

**Comment by CodyBerenson** [comment]
hello my friend Jason.  Glad to hear from you.  :)  I've been busy matching Sean Cody.  My life is on hold as I match all these libraries.  LOL.  Since I've had a couple bourbons tonight, I"ll work on the pull request tomorrow.  I don't want Aiden to get mad if the bourbon steps on version control.  LOL!

You're right, that looks much cleaner.  The main agents screen was getting a bit cluttered.  

Everyone's preferences are probably vastly different.  Me?  I set up a library per studio (for scenes) and a master library for features.  So, for me, it makes COMPLETE sense to separate the blog agents for scenes from the feature agents....  I don't really see myself using studio specific scrapers....but, perhaps i just haven't found a studio in my collection that isn't well represented on waybig or fagalicious...

good to catch up.

Will i run into you in Rotterdam in May for ESC?  :)

**Comment by JPH71** [comment]
Hi cody

Hold fire on creating the pull requests... I forgot to up the version
numbers... I will do that this evening when I get back home from work.
Could you think of appropriate names for the split.. I have Gay Blog Agents
and Gay Film Agents...
Under Gay Blog - we will have Waybig and Fagalicious
Under Gay Films - we will have the rest

I will probably be motivated if a number of people at work get their act
together and organise a Eurovision trip to Rotterdam!
So hopefully see you there!


Jason

**Comment by CodyBerenson** [comment]
LOL.  Douze Points!

So, I'm not sure if "Blog" is universally intuitive for folks, expecially newbies, that its for Scenes with DVD quality coverart......  perhaps incorporate the word Scenes.  "Gay Scenes Blog"....what do you think?

**Comment by JPH71** [comment]
I like the scenes...
So Gay Scenes and Gay Films
I will start working on that xx

**Comment by JPH71** [comment]
Hi Cody

This is what I have done:
1. GayAdult - Both Blogs and VOD sites (For libaries that have combos of full movies and scenes)
      This works as the previous one did
2. GayAdultFilms - Just VOD sites - like AEBN, GayHotmovies etc
3. GayAdultScenes - Just Blog sites

I have made changes to the scrappers so that they contribute to GayAdult as well as GayAdultFilms or GayAdultScenes. for example:

- AEBNii - contributes to GayAdult and GayAdultFilm, 
- Waybig contributes to GayAdult and GayAdultScenes

I found another website CD Universe and created a scrapper for it..

So these are New:
1. Gay Adult Films
2. Gay Adult Scenes
3. CD Universe

These are Reworked:
1. Gay Adult (reworked)
2. AEBNii
3. Fagalicious
4. GayDVDEmpire
5. GayHotMovies
6. GEVI
7. HomoActive
8. WayBig

All Agents have the delay preference now - so one can delay requests by 0, 10, 20, 30, 40, 50 or 60 seconds.

I am still trying to find out how to set cookies, or click on the enter button for the following:
- TLAGay
- Gay VOD Club
- MaleFlixxTV

If you could do the pull request thingy, I will be eternally grateful.
I have also improved the info windows for GayAdult, GayAdultFilms and GayAdultScenes like thus:
![image](https://user-images.githubusercontent.com/43102249/74095017-8ef97980-4ae2-11ea-8378-a07e72c68e56.png)
![image](https://user-images.githubusercontent.com/43102249/74095027-ac2e4800-4ae2-11ea-96cd-fa3f69894967.png)
![image](https://user-images.githubusercontent.com/43102249/74095031-bea88180-4ae2-11ea-9d36-3dcf54905a93.png)

The links within - take the user to our github page.

Here are the bundles:
[GayAdultFilms.bundle.zip](https://github.com/acvigue/pgma/files/4176388/GayAdultFilms.bundle.zip)
[GayAdultScenes.bundle.zip](https://github.com/acvigue/pgma/files/4176389/GayAdultScenes.bundle.zip)
[GayDVDEmpire.bundle.zip](https://github.com/acvigue/pgma/files/4176390/GayDVDEmpire.bundle.zip)
[GayHotMovies.bundle.zip](https://github.com/acvigue/pgma/files/4176391/GayHotMovies.bundle.zip)
[GEVI.bundle.zip](https://github.com/acvigue/pgma/files/4176392/GEVI.bundle.zip)
[HomoActive.bundle.zip](https://github.com/acvigue/pgma/files/4176393/HomoActive.bundle.zip)
[WayBig.bundle.zip](https://github.com/acvigue/pgma/files/4176394/WayBig.bundle.zip)
[AEBNii.bundle.zip](https://github.com/acvigue/pgma/files/4176395/AEBNii.bundle.zip)
[CDUniverse.bundle.zip](https://github.com/acvigue/pgma/files/4176396/CDUniverse.bundle.zip)
[Fagalicious.bundle.zip](https://github.com/acvigue/pgma/files/4176397/Fagalicious.bundle.zip)
[GayAdult.bundle.zip](https://github.com/acvigue/pgma/files/4176398/GayAdult.bundle.zip)

**Comment by JPH71** [comment]
Let me know if you find owt wrong when you are testing....
Many thanks

Jason

**Comment by JPH71** [comment]
Hey Cody

Found an error in Gevi, too exuberant with pasting....
her is the corrected version
[GEVI.bundle.zip](https://github.com/acvigue/pgma/files/4178153/GEVI.bundle.zip)

I will be reworking gayDVDEmpire next as it could be made more efficient...

**Comment by CodyBerenson** [comment]
Hey Jason,

My observations:

1.  I do really like the separation of the scenes agents from the films agents.  So much cleaner.
2.  Personally, I removed the "GayPornCollector" and "CockPorn" bundles so I no longer have a "Gay Adult" set of agents.  But I do understand plenty of folks may comingle libraries so its an important option for them.
3.  I've tried a FEW movies against the new CDUniverse agent and all good so far.   I'll add it to my rotation.  :)

As always, you're AMAZING.  :)

Hope you're well.

**Comment by JPH71** [comment]
All good here...
I need to get sorting the rest of my Tim tales against fagalicious as
waybig dies not have any of the old ones...
Still battling with the cookies to enable to bypass 18+ enter buttons on
makeflixxtv and gayVodclub...
Miserable windy day here today as was the weekend... Beautifull programming
weather!
Hope you are keeping well too...

**Comment by CodyBerenson** [comment]
Hey Jason my friend.  Not sure if you've ever checked out QueerClick.com.  It is like waybig and fagalicious, and there are plenty of times where files aren't indexed at either W or F but they are at QC.  The challenge is that the coverart is oversized, and the default way plex pulls in cover art oddly crops.....  Not sure if there's any way, should you build a QC scraper, to crop the coverart on the fly.  For example when i grab coverart from QC, i crop it thusly:
![image](https://user-images.githubusercontent.com/40373801/74302550-ebe47200-4d24-11ea-9a33-9f38c58f5cee.png)

**Comment by JPH71** [comment]
I will check that one out...
It's good to get another challenge 😆👍

**Comment by CodyBerenson** [comment]
Hey Jason, just noticed.  The new CDUniverse scraper has the release-date issue that we saw with Fagalicious.  For example:

https://www.cduniverse.com/productinfo.asp?pid=12356207&style=gay

This film has a release date of January 23, 2020, but the scraper populates the release date metadata as December 31, 2020.  

Cheers!

**Comment by JPH71** [comment]
I will look at that today ….
Thanks for noticing that.

Jason

**Comment by JPH71** [comment]
Hi Cody

I am not experiening the error with the dates... this is what i got when i tried the movie title you sent

![image](https://user-images.githubusercontent.com/43102249/74526892-4ac00c00-4f1c-11ea-8af8-d7aef3717755.png)

Though I will make an improvement, I just realised that one can scrape collection information from CD Univerese... as it places it together with the genres

**Comment by JPH71** [comment]
Cody / Aiden
I am still looking inot image cropping...
It can be done but one has to import extra python libraries..

Aiden, - does plex allow you to add python import libraries as a resource to a scrapper... if it does can you tell me how to do it?

Also I wanted to forward you the code for GayVODClub as I can not seem to get the Cookies set correctly... any help would be appreciated

Cheers

Jason

**Comment by CodyBerenson** [comment]
Hi Jason, interesting re: the dates....i'll keep an eye out on other CDUniverse matches.  could have just been a fluke!

**Comment by JPH71** [comment]
Working on queerclick... Hopefully get it all sorted in 2 days xxx

**Comment by CodyBerenson** [comment]
That's terrific Jason.  :)  However, I hope you got to enjoy the weekend.  It is a holiday here in the USA, so a nice (but cold) long weekend for us.  :)

**Comment by JPH71** [comment]
You lucky buggers... Any holiday us better than nowt...
No internet tonight ..... Greater
Xxx

**Comment by CodyBerenson** [comment]
when one posts pics that break the internet....the internet breaks.  cause.  effect.  physics 101.

**Comment by CodyBerenson** [comment]
Hey Jason, happy end of the week!  Hope you enjoyed your week and get to rest up and have some fun this weekend!  Not too much fun though.  No international incidents, please.

**Comment by JPH71** [comment]
Hey Cody
At last, the weekend... All nice and quiet here x

**Comment by CodyBerenson** [comment]
Hey Jason.  should we close this thread and open one for QueerClick, since that's one of the next agents in the works?  I tell you, between your fagalicious and waybig and a future queerclick, virtually every scene from most collections are covered.  :)

**Comment by JPH71** [comment]
I was going to suggest that the other day.
I am working on queerclick at the moment x

**Comment by CodyBerenson** [comment]
ok.  i'll close this, open queer click.  and we can reconcile there all the latest updates you have to all of the agents, so if any need to be added to the master, i'll open a pull request.

**Comment by JPH71** [comment]
Perfect..
You are a star as always x
